R/FilterStateLogical.R

```diff
@@ -163,11 +168,7 @@ LogicalFilterState <- R6::R6Class( # nolint
       do.call(super$initialize, args)
 
       private$set_choices(choices)
-      if (is.null(selected)) {
-        private$set_selected(TRUE)
-      } else {
-        private$set_selected(selected)
-      }
```

Comment:
I am not sure this version is better than the previous one. For logical values, the default selection is  `TURE`, not the alphabetically first one, which would be `FALSE`.

Reply:
In the definition of the remove_out_bound function, I have set the default value to TRUE. You can find the code at this link: [GitHub link](https://github.com/insightsengineering/teal.slice/blob/279_multiple%40filter_panel_refactor%40main/R/FilterStateLogical.R#L250).

The logic of this function is similar to the remove_out_of_bound() function in filterStatechoice.R, with the only difference being that instead of using choice[1], I have set it to TRUE.